packages/core/state/interaction/logics.ts

```diff
@@ -582,17 +582,81 @@ const setIsSmallScreen = createLogic({
  * Logs details of files that are being moved.
  */
 const moveFilesLogic = createLogic({
-    async process({ action, getState }: ReduxLogicDeps, _dispatch, done) {
+    async process({ action, getState }: ReduxLogicDeps, dispatch, done) {
         try {
             const httpFileService = interactionSelectors.getHttpFileService(getState());
             const username = interactionSelectors.getUserName(getState());
-            const fileIds = (action as MoveFilesAction).payload.fileDetails.map((file) => file.id);
-            const cacheStatuses = await httpFileService.cacheFiles(fileIds, username);
 
-            // TODO: What to do with the status
-            console.log("Cache statuses:", cacheStatuses);
+            const fileDetails = (action as MoveFilesAction).payload.fileDetails;
+
+            // Map file IDs to file names for easy lookup
+            const fileIdToNameMap = Object.fromEntries(
+                fileDetails.map((file) => [file.id, file.name])
+            );
+
+            // Extract file IDs
+            const fileIds = fileDetails.map((file) => file.id);
+
+            const response = await httpFileService.cacheFiles(fileIds, username);
+            const cacheStatuses = response.cacheFileStatuses;
+
+            // Check if the response is empty.
+            if (!cacheStatuses || Object.keys(cacheStatuses).length === 0) {
+                dispatch(
+                    interaction.actions.processError(
+                        "moveFilesNoFilesProcessed",
+                        "No files were processed. Please check the request or try again."
+                    )
+                );
+                return;
+            }
+
+            const successfulFiles: string[] = [];
+            const failedFiles: string[] = [];
+
+            Object.entries(cacheStatuses).forEach(([fileId, status]) => {
+                if (
+                    status === "DOWNLOAD_COMPLETE" ||
+                    status === "DOWNLOAD_IN_PROGRESS" ||
+                    status === "DOWNLOAD_STARTED"
+                ) {
+                    successfulFiles.push(fileId);
+                } else if (
+                    status === "FILE_RECORD_NOT_FOUND" ||
+                    status === "FILE_NOT_FOUND_ON_CLOUD"
+                ) {
+                    failedFiles.push(fileId);
+                }
+            });
+
+            // Dispatch net success message. (Assuming some files were successful)
+            if (successfulFiles.length > 0) {
+                dispatch(
+                    interaction.actions.processSuccess(
+                        "moveFilesSuccess",
+                        `${successfulFiles.length} out of ${
+                            Object.keys(cacheStatuses).length
+                        } files were successfully cached.`
```

Comment:
"were successfully cached" is maybe misleading since this just kicks off the download process or refreshes it. I think something like "files will be available in the NAS after downloads finish asynchronously".

Do you have a message about how the downloads will become available in the modal that appears after clicking the copy option in the context menu?

Reply:
I totally agree with the new wording. In regards to when the files will be available, I do not have any information for users. We do know that files are given a 180-day lease which there is a note about in the modal.

Reply:
Resolved wording in [3726ec5](https://github.com/AllenInstitute/biofile-finder/pull/350/commits/3726ec57d03eed9085095800cc3a287af3a29398)

Reply:
> files were successfully cached. Files will be available in the NAS (VAST) after downloads finish asynchronously

This reads like the "cache" & "NAS" are different things (as in we have past tense cached it, but it still needs downloading to the NAS). Maybe just keep it vague like: 

"files were successfully queued for download to NAS (Vast) from cloud. Files will be available in the NAS after downloads finish asynchronously"

Reply:
resolved in [60cf69a](https://github.com/AllenInstitute/biofile-finder/pull/350/commits/60cf69a2a3c544e763e29152194146d5022d176e)